### DISCUSSION TITLE
How to bind server to localhost only?

### DISCUSSION THREAD

**Discussion by mbiette** [atoti/atoti#590] [Q&A] [answered]
Hello,

When opening a session it binds to all the ip addresses of the machine by default making it accessible to anybody, which I would like to restrict.

Example creating a very simple Session:
```
$ python
>>> import atoti
Welcome to atoti 0.6.5!
[...]
>>> s = atoti.Session(name="test", port=5111)
>>>
```
It actually listens to all incoming connections:
```
tcp        0      0 0.0.0.0:5111            0.0.0.0:*               LISTEN
```
but what I'm trying to get is:
```
tcp        0      0 127.0.0.1:5111            0.0.0.0:*               LISTEN
```

The only part of the documentation mentioning a url_pattern with localhost is:
https://docs.atoti.io/latest/releases/0.6.0.html#id5
but the Session class doesn't take a url or url_pattern parameter.

Is there a documentation page or configuration that could help me?

Thank you and best regards

**Comment by mailys-ds** [comment]
Hi Maxime,
what's the goal of restricting the atoti session to only 127.0.0.1, how do you make it accessible to other users then? Or is the issue that you're working on a personal machine on a private network and would like the other computers on your network to not be able to access your session?

**Reply by mbiette** [reply]
Hello Mailys,

Yes, I would like other computers on my network to not be able to access the session. In regards to making it accessible, this is something I can "add" in a second time.

**Comment by mailys-ds** [comment, marked as answer]
To change the IP address of your session, you can add an option in the `java_options` when creating the session. Here is the configuration you need:
```python
import atoti as tt
s = tt.Session(name="test", port=5111, java_options=["-Dserver.address=127.0.0.1"])
```
Let me know if that works for you.

**Reply by mbiette** [reply]
Wonderful, it works perfectly. Thank you!
